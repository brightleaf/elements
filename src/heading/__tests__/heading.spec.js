import React from 'react'
import { render, cleanup } from '@testing-library/react'
import {
  toHaveClass,
  toHaveAttribute,
  toContainElement,
} from '@testing-library/jest-dom'
import { Heading } from '../'

expect.extend({ toHaveClass, toHaveAttribute, toContainElement })
describe('The Heading Component', () => {
  afterEach(cleanup)
  it('should render a div with the class heading', () => {
    const { container } = render(<Heading> </Heading>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('heading')
  })
  it('should render a div with the class heading and the className passed in', () => {
    const { container } = render(<Heading className="heading-test" />)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('heading')
    expect(container.firstChild).toHaveClass('heading-test')
  })
  it('should render a p with the class heading when as property is set to `div`', () => {
    const { container } = render(<Heading as="div" />)
    expect(container).toMatchSnapshot()

    expect(container.firstChild).toHaveClass('heading')
    const p = document.querySelector('div.heading')
    expect(container).toContainElement(p)
  })
})
