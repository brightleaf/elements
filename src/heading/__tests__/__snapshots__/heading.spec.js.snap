// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Heading Component should render a div with the class heading 1`] = `
<p
  class="heading"
>
   
</p>
`;

exports[`The Heading Component should render a div with the class heading and the className passed in 1`] = `
<p
  class="heading heading-test"
/>
`;

exports[`The Heading Component should render a p with the class heading when as property is set to \`div\` 1`] = `
<div>
  <div
    class="heading"
  />
</div>
`;
